### PR TITLE
Android - Optimization

### PR DESCRIPTION
**Pull request by AndrewEQ**
- added options for optimization:

```
"Android": {
    "MultiDexEnabled":true,
    "Optimize": {
      "MinifyEnabled": true,
      "ShrinkResources": true,
      "DefaultAndroidProGuard": "proguard-android-optimize.txt",
      "PrependRulesFile": "proguard-rules-prepend.pro",
      "AppendRulesFile": "proguard-rules-append.pro"
    },
    ...
```

Default android proguard rules(proguard-android.txt) are applied when not specified
```
"Android": {
    "Optimize": {
      "MinifyEnabled": true,
      "ShrinkResources": true
    },
    ...
```

Can add custom proguard rules before or after the main generated class rules:
```
"Android": {
    "Optimize": {
      "MinifyEnabled": true,
      "ShrinkResources": true,
      "PrependRulesFile": "proguard-rules-prepend.pro",
      "AppendRulesFile": "proguard-rules-append.pro"
    },
    ...
```

Create "proguard-rules-prepend.pro" in root of Project folder:

e.g.

MyApp/MainView.ux
MyApp/proguard-rules-prepend.pro

...

proguard-rules-prepend.pro
```
-keepattributes *Annotation*
-keepattributes Exceptions
-keepattributes InnerClasses
-keepattributes Signature
-keepattributes SourceFile,LineNumberTable
```

### CODE REVIEW

**Comment by mortend** [comment]
Just pushed #266 to `master`. If you rebase you should get your compiler generated `app/proguard-rules.pro`. Cheers! :)

**Comment by AndrewEQ** [comment]
Shweet, I'll try it out now now...

File: lib/UnoCore/Targets/Android/Android.uxl [hunk 1]
```diff
@@ -238,6 +238,7 @@
     <Set Commands.Run="@(HOST_WIN32:Defined:Test('run.bat', 'bash run.sh'))" />
 
     <ProcessFile Name="app/build.gradle" />
+    <ProcessFile Name="app/proguard-rules.pro" />
```
Comment (mortend) [inline lib/UnoCore/Targets/Android/Android.uxl]: This file is compiler-generated and shouldn't be listed here.

Comment (AndrewEQ) [reply lib/UnoCore/Targets/Android/Android.uxl]: Is there a way to merge and or append to the compiler-generated one if a user-defined one is defined?

Comment (AndrewEQ) [reply lib/UnoCore/Targets/Android/Android.uxl]: Got an update coming 😄 ...

**Comment by AndrewEQ** [comment]
Aww, how do kick start travis again?

**Comment by mortend** [comment]
> Aww, how do we kick start travis again?

There's a "Restart build" button - restarted now.

**Pull request closed without merging**